prog/dftb+/lib_dftb/rangeseparated.F90

```diff
@@ -177,15 +181,15 @@ subroutine initAndAllocate(this, nAtom, hubbu, species, screen, omega, rsAlg, tS
       !> Algorithm for range separation
       integer, intent(in) :: rsAlg
 
-      !> Is this spin restricted (F) or unrestricted (T)
-      logical, intent(in) :: tSpin
+      !> Is this spin restricted (1) or unrestricted (2)
+      integer, intent(in) :: nSpin
 
       this%tScreeningInited = .false.
       this%pScreeningThreshold = screen
       this%omega = omega
       this%lrEnergy = 0.0_dp
       this%rsAlg = rsAlg
-      this%tSpin = tSpin
+      this%nSpin = nSpin
```

Comment:
Dangerous! `nSpin` can be also `4` (non-collinear spin), which the routines here can not handle. You can keep `this%nSpin`, but please set it up using `tSpin` (and not `nSpin) accordingly. `tSpin` is only `.true.` for collinear spin and `.false.` otherwise (no spin polarisation, non-collinear spin).